Added ROS2 introduction (#60)
eProsima/vulcanexus: docs/rst/introduction/introduction.rst

.. _vulcanexus_global_introduction:

Vulcanexus Overview
===================

*Vulcanexus*, the *ROS 2* all-in-one tool set, expands the *ROS 2* environment improving the developer experience with *ROS 2* providing open source features and tools not available with the current `ROS 2 release <https://www.ros.org/reps/rep-2000.html#release-schedule>`_.
Consequently, *Vulcanexus* shares the underlying concepts, principles and architecture of :doc:`ROS 2 <../../ros2_documentation/source/index>`.

> For more information about *ROS 2* please refer to the 2022 Science Robotics vol. 7 (May 2022) paper `Robot Operating System 2: Design, architecture, and uses in the wild <https://www.science.org/doi/10.1126/scirobotics.abm6074>`_ by S. Macenski, T. Foote, B. Gerkey, C. Lalancette and W. Woodall.

Reviewer: I'd include this as a footnote. Check [here](https://devopstutodoc.readthedocs.io/en/stable/documentation/doc_generators/sphinx/rest_sphinx/footnotes.html) how to do so

Consequently, only patch releases that fix bugs without modifying API and ABI are considered for updates in the *ROS 2* environment.
*Vulcanexus*, on the other hand, updates the middleware layer any time *eProsima Fast DDS* releases a new version in order to benefit from the latest features and fixes.
*Vulcanexus* takes charge of releasing a new binary stable distribution that is compatible with the potential ABI breaks so the *Vulcanexus* user has only to update its *Vulcanexus* distro to benefit from the latest *Fast DDS* release.
For instance, *Vulcanexus* provides :ref:`PKCS #11 security support <tutorials_security_pkcs11_pkcs11>` whereas *ROS 2* does not support it (as of September 2022).

Algorithms
^^^^^^^^^^

Another category includes robotic application's algorithms.
*Vulcanexus*, being a *ROS 2* all-in-one tool set, provides the algorithms included in *ROS 2* ecosystem.

Developer tools
^^^^^^^^^^^^^^^

Finally, *ROS 2* provides several development tools for debugging, logging, visualization, introspection, simulation, etc.
*Vulcanexus* adds to *ROS 2* toolset including several tools:

> - `eProsima Fast DDS Monitor <https://fast-dds-monitor.readthedocs.io/en/latest/>`_: graphical desktop application to monitor DDS communications when using *eProsima Fast DDS*.

Reviewer: ```suggestion
- `ROS 2 Monitor <https://fast-dds-monitor.readthedocs.io/en/latest/>`_: graphical desktop application to monitor ROS 2 communications.
```

> - `eProsima Shapes Demo <https://eprosima-shapes-demo.readthedocs.io/en/latest/index.html>`_: first demo application to understand the most used DDS Quality of Service (QoS) and test DDS and *ROS 2* communication.

Reviewer: ```suggestion
- `ROS 2 Shapes Demo <https://eprosima-shapes-demo.readthedocs.io/en/latest/index.html>`_: first demo application to understand the most used ROS 2 Quality of Service (QoS) and test DDS and *ROS 2* communication.
```

- :ref:`micro-ROS <tutorials_micro_introduction_microros>`: default embedded toolkit for *ROS 2* and *Vulcanexus*.

> - `eProsima DDS Router <https://eprosima-dds-router.readthedocs.io/en/latest/index.html>`_: end-user software application enabling the connection of distributed DDS networks.

Reviewer: ```suggestion
- `ROS 2 Router <https://eprosima-dds-router.readthedocs.io/en/latest/index.html>`_: end-user software application enabling the connection of distributed ROS 2 environments.
```

.. _vulcanexus_global_introduction:

Vulcanexus Overview
===================

*Vulcanexus*, the *ROS 2* all-in-one tool set, expands the *ROS 2* environment improving the developer experience with *ROS 2* providing open source features and tools not available with the current `ROS 2 release <https://www.ros.org/reps/rep-2000.html#release-schedule>`_.
Consequently, *Vulcanexus* shares the underlying concepts, principles and architecture of :doc:`ROS 2 <../../ros2_documentation/source/index>`.
For more information about *ROS 2* please refer to the `Robot Operating System 2: Design, architecture, and uses in the wild <https://www.science.org/doi/10.1126/scirobotics.abm6074>`_ [#]_.
This section will follow and summarize the most important points contained in this paper, pointing out the differences and improvements that *Vulcanexus* provides.

> .. [#] S. Macenski, T. Foote, B. Gerkey, C. Lalancette and W. Woodall, "Robot Operating System 2: Design, architecture, and uses in the wild", Science Robotics vol. 7, May 2022

Reviewer: I'd move this to the end of the page (footnote) as this page is short enough

Reviewer 2: Done in 0d61e2fc8bfb6587e04c98238b18414c92b2c0b2
